homeassistant/components/axis/diagnostics.py

```diff
@@ -0,0 +1,46 @@
+"""Diagnostics support for Axis."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.diagnostics import async_redact_data
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_MAC, CONF_PASSWORD
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN as AXIS_DOMAIN
+
+REDACT_CONFIG = {CONF_MAC, CONF_PASSWORD}
+REDACT_BASIC_DEVICE_INFO = {"SerialNumber", "SocSerialNumber"}
+REDACT_VAPIX_PARAMS = {"root.Network", "System.SerialNumber"}
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, config_entry: ConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+    device = hass.data[AXIS_DOMAIN][config_entry.unique_id]
+    diag: dict[str, Any] = {}
+
+    diag["config"] = async_redact_data(config_entry.data, REDACT_CONFIG)
+    diag["options"] = dict(config_entry.options)
```

Comment:
you can just provide the whole config entry, which also adds some additional info like the system settings and if the entry is created by the user, or by discovery...
 ```suggestion
     diag["config_entry"] = async_redact_data(config_entry.as_dict(), REDACT_CONFIG)
```

Reply:
Changed according to your suggestion